atef/bin/main.py

```diff
@@ -14,34 +14,35 @@
 DESCRIPTION = __doc__
 
 
-MODULES = ("help", )
+COMMAND_TO_MODULE = {
+    "check": "check",
+}
 
 
-def _try_import(module):
-    relative_module = f'.{module}'
-    return importlib.import_module(relative_module, 'atef.bin')
+def _try_import(module_name):
+    return importlib.import_module(f".{module_name}", 'atef.bin')
```

Comment:
What exactly is going on here? Why are we not importing these modules the normal way?

Reply:
This is the "easily support various subcommands through a simpleish interface" thing I used in a bunch of projects and added to the cookiecutter: https://github.com/pcdshub/cookiecutter-pcds-python/blob/master/%7B%7B%20cookiecutter.folder_name%20%7D%7D/%7B%7B%20cookiecutter.import_name%20%7D%7D/bin/main.py

The short of it is: add your module name to the list, add functions for creating the arg parser + main, and you're done. If it's not importable for some reason (optional dependencies? other import-related error?) it'll still allow you to use other subcommands.

Reply:
Alright, seems ok. I just found it really confusing/strange and I don't expect any linters to be able to pick up on typos in the COMMAND_TO_MODULE dict the same way they would for something much more straightforward e.g.
```
from . import check
from . import grafana

SUBMODULES = (check, grafana)

COMMANDS = {
    module.__name__: module.build_arg_parser
    for module in SUBMODULES
}
```

Reply:
I think that was the original implementation in pytmc, but then sometimes pyqt5 wasn't installed - which made every pytmc subcommand (even those that didn't require pyqt5) inaccessible. 

I think I could be convinced the benefits of the static checker outweigh the benefits of the current implementation. Try/catch on import, don't add it to the commands dict. I'm only not sure about how to let the user know the reason that the subcommand isn't available without being too annoying.

It may be worth revisiting, but I think probably outside of the scope of the PR. (And in scope of 4ish other projects that use the same structure 😁 )

Reply:
Yeah, I mean, it confused me but if it does the thing it does the thing